run--02a--run-and-validate-SAE.Rmd

```diff
@@ -945,6 +891,533 @@ sae_results_incpov_ctrls_mix_05$sae_out %>%
+
+1. aligning our SAE estimates for each income-to-poverty category to the ACS 1-year 5% sample values for each PUMA, and
+2. expanding the variance of the SAE estimates to equal the variance of ACS 5-year analogs.
+
+In this way, we anchor the distribution of our new estimates in the most reliable and most recent values that are available.
+
+```{r calculate variance of acs5 analogs to each income-to-poverty measure}
+# /!\ Consider moving this to run--01d
+acs5_incpov_cat <- 
+  acs5tract %>% 
+  select(matches("GEOID|incpov_(lt6|6to11).+_est")) %>% 
+  pivot_longer(cols = -GEOID) %>% 
+  mutate(age   = str_replace(name, "incpov_(.+)_r.+", "\\1"),
+         range = str_replace(name, ".+\\d_(r.+)_est", "\\1")) %>% 
+  select(-name) %>% 
+  pivot_wider(names_from = "range",
+              values_from = "value") %>% 
+  mutate(r50to100 = r0to50 + r50to100) %>% 
+  pivot_longer(cols = -c(GEOID, age)) %>% 
+  # Put values into format that will merge into SAE `vc_value` field names
+  mutate(vc_value = 
+           name %>% 
+           str_replace("r(\\d+)to(\\d+)", "\\1%-\\2%") %>% 
+           str_replace("r(\\d+)plus", "\\1%+") %>% 
+           str_replace("199", "200") %>% 
+           str_replace("299", "300") %>% 
+           str_replace("399", "400"))
+
+acs5_incpov_cat_var <- 
+  acs5_incpov_cat %>% 
+  merge(geo_crosswalk %>% select(GEOID, PUMA) %>% unique(),
+        by = "GEOID",
+        all.x = TRUE) %>% 
+  group_by(PUMA, age, vc_value) %>% 
+  # /!\ Consider whether to build this as a population-weighted variance
+  summarize(acs5_tract.in.puma_sd = sd(value, na.rm = TRUE)) %>% 
+  ungroup()
+```
+
+
+```{r calculate variance of acs5 analogs to each income-to-poverty measure}
+pop_long <- 
+  pop_by_age %>% 
+  #filter(year == base_year) %>% 
+  pivot_longer(cols = -c(GEOID, year)) %>% 
+  mutate(age = str_replace(name, ".+_(\\d+to\\d+).+", "\\1")) %>% 
+  select(GEOID, year, age, pop = value)
+
+```
+
+```{r prep acs1_5pct data for merge}
+acs1_5pct_puma_formerge <- 
+  acs1_5pct_puma %>% 
+  pivot_longer(cols = -c(GEOID, PUMA, age)) %>% 
+  filter(!str_detect(name, "_n$")) %>% 
+  mutate(vc_value =
+           name %>% 
+           str_replace(".+_(\\d+)_(\\d+).+", "\\1%-\\2%") %>% 
+           str_replace(".+agg_(\\d+)_pct", "\\1%+")) %>% 
+  rename(acs1_5pct_pct = value)
+```
+
+
+```{r misc functions for use in redistribution}
+bound <- function(x, lower = 0, upper = 1) {
+  x[x < lower] <- lower
+  x[x > upper] <- upper
+  return(x)
+}
+
+if (FALSE) {
+  bound(c(-1, 0.2, 0.8, 5))
+}
+```
+
+
+```{r function for performing redistribution}
+# Values for easy interactive debugging
+if (FALSE) {
+  sae_to_redis <- sae_results_incpov_ctrls_100_05
+  sae_age <- "05" # "612"
+}
+
+redis_sae <- function(sae_to_redis, sae_age) {
+  
+  pop_age <- case_when(sae_age == "05"  ~ "0to5", 
+                       sae_age == "612" ~ "6to12")
+  acs_age <- case_when(sae_age == "05"  ~ "lt6", 
+                       sae_age == "612" ~ "6to11")
```

Comment:
I'm assuming this is related to the earlier comment for line 204, is there any reason we do `6to11` instead of `6to12`?